cpp/examples/parquet_io/parquet_io.hpp

```diff
@@ -0,0 +1,162 @@
+    {"NONE", compression_type::NONE},
+    {"AUTO", compression_type::AUTO},
+    {"SNAPPY", compression_type::SNAPPY},
+    {"BZIP2", compression_type::BZIP2},
+    {"BROTLI", compression_type::BROTLI},
+    {"ZIP", compression_type::ZIP},
+    {"XZ", compression_type::XZ},
+    {"ZLIB", compression_type::ZLIB},
+    {"LZ4", compression_type::LZ4},
+    {"LZO", compression_type::LZO},
+    {"ZSTD", compression_type::ZSTD}};
+
+  std::transform(name.begin(), name.end(), name.begin(), ::toupper);
+  if (map.find(name) != map.end()) { return map.at(name); }
+  throw std::invalid_argument("FATAL: " + std::string(name) +
+                              " is not a valid compression type.\n\n"
+                              "Available compression_type types: NONE, AUTO, SNAPPY,\n"
+                              "BZIP2, BROTLI, ZIP, XZ, ZLIB, LZ4, LZO, ZSTD\n"
+                              "\n"
+                              "Exiting...\n");
+}
+
+/**
+ * @brief Get the optional page size stat frequency from they keyword
+ *
+ * @param use_stats keyword affirmation string such as: Y, T, YES, TRUE, ON
+ * @return optional page statistics frequency set to full (STATISTICS_COLUMN)
+ */
+[[nodiscard]] std::optional<cudf::io::statistics_freq> get_page_size_stats(std::string use_stats)
+{
+  std::transform(use_stats.begin(), use_stats.end(), use_stats.begin(), ::toupper);
+
+  // Check if the input string matches to any of the following
+  if (not use_stats.compare("ON") or not use_stats.compare("TRUE") or
+      not use_stats.compare("YES") or not use_stats.compare("Y") or not use_stats.compare("T")) {
+    // Full column and offset indices - STATISTICS_COLUMN
+    return std::make_optional(cudf::io::statistics_freq::STATISTICS_COLUMN);
+  }
+
+  return std::nullopt;
+}
+
+/**
+ * @brief Light-weight timer for parquet reader and writer instrumentation
+ *
+ * Timer object constructed from std::chrono, instrumenting at microseconds
+ * precision. Can display elapsed durations at milli and micro second
+ * scales. Timer starts at object construction.
+ */
+class Timer {
+ public:
+  using micros = std::chrono::microseconds;
+  using millis = std::chrono::milliseconds;
+
+  Timer() { start(); }
+  ~Timer() { stop(); }
+
+  void start() { start_time = std::chrono::high_resolution_clock::now(); }
+  void stop() { end_time = std::chrono::high_resolution_clock::now(); }
+
+  auto duration_us() { return std::chrono::duration_cast<micros>(end_time - start_time).count(); }
+  auto duration_ms() { return std::chrono::duration_cast<millis>(end_time - start_time).count(); }
+
+  void print_elapsed_micros() { std::cout << "Elapsed Time: " << duration_us() << "us\n\n"; }
+  void print_elapsed_millis() { std::cout << "Elapsed Time: " << duration_ms() << "ms\n\n"; }
+
+ private:
+  using time_point_t = std::chrono::time_point<std::chrono::high_resolution_clock>;
+  time_point_t start_time;
+  time_point_t end_time;
```

Comment:
fairly nitpicky given that this is sample code, but I feel like `Timer` could be safer/simpler to use.
```suggestion
  Timer() { reset(); }

  void reset() { start_time = std::chrono::high_resolution_clock::now(); }

  auto elapsed() { return (std::chrono::high_resolution_clock::now() - start_time).count(); }

  void print_elapsed_micros() { std::cout << "Elapsed Time: " << std::chrono::duration_cast<micros>(elapsed()) << "us\n\n"; }
  void print_elapsed_millis() { std::cout << "Elapsed Time: " << std::chrono::duration_cast<millis>(elapsed()) << "ms\n\n"; }

 private:
  using time_point_t = std::chrono::time_point<std::chrono::high_resolution_clock>;
  time_point_t start_time;
```

Reply:
All done. Thank you for the suggestion.